backend/src/federation/index.ts

```diff
@@ -36,19 +32,61 @@ export const startDHT = async (
     logger.info(`keyPairDHT: publicKey=${keyPair.publicKey.toString('hex')}`)
     logger.debug(`keyPairDHT: secretKey=${keyPair.secretKey.toString('hex')}`)
 
+    const apiList = {
+      apiVersions: Object.values(ApiVersionType).map(function (apiEnum) {
+        return { api: apiEnum, url: CONFIG.FEDERATION_COMMUNITY_URL }
+      }),
+    }
+    logger.debug(`ApiList: ${JSON.stringify(apiList)}`)
+
     const node = new DHT({ keyPair })
 
     const server = node.createServer()
 
     server.on('connection', function (socket: any) {
       // noiseSocket is E2E between you and the other peer
       // pipe it somewhere like any duplex stream
-      logger.info(`Remote public key: ${socket.remotePublicKey.toString('hex')}`)
+      logger.info(`server on... with Remote public key: ${socket.remotePublicKey.toString('hex')}`)
       // console.log("Local public key", noiseSocket.publicKey.toString("hex")); // same as keyPair.publicKey
 
-      socket.on('data', (data: Buffer) => logger.info(`data: ${data.toString('ascii')}`))
-
-      // process.stdin.pipe(noiseSocket).pipe(process.stdout);
+      socket.on('data', async (data: Buffer) => {
+        try {
+          logger.info(`data: ${data.toString('ascii')}`)
+          const json = JSON.parse(data.toString('ascii'))
+          if (
+            json.apiVersions &&
+            Array.isArray(json.apiVersions) &&
+            json.apiVersions.length > 0 &&
```

Comment:
This check can be reduced to `isArray(json)` if no senseless field `apiVersions` is submitted.
The array length check is not needed at all - if the length is zero the for loop is executed zero times.

Reply:
removed and unnecessary by reintroducing the type definitions, see above